src/NHibernate/Cfg/ConfigurationManagerBase.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Configuration;
+using System.Linq;
+using NHibernate.Cfg.ConfigurationSchema;
+
+namespace NHibernate.Cfg
+{
+	/// <summary>
+	/// Base class for configuration settings that NHibernate might require.
+	/// </summary>
+	public abstract class ConfigurationManagerBase
```

Comment:
Lets rename this class to `ConfigurationProvider` and move each class into separate file.

Reply:
Should all related classes/properties be renamed accordingly too?

Reply:
Yes